static/js/stream_edit.js

```diff
@@ -583,6 +585,66 @@ export function initialize() {
         settings_ui.do_settings_change(channel.patch, url, data, status_element);
     }
 
+    $("#manage_streams_container").on("click", ".copy_email_button", (e) => {
+        const stream_id = get_stream_id(e.target);
+        const stream = sub_store.get(stream_id);
+        let address = stream.email_address;
+
+        const copy_email_address = render_copy_email_address_modal({
+            email_address: address,
+        });
+
+        dialog_widget.launch({
+            html_heading: $t_html({defaultMessage: "Configure and copy email address"}),
+            html_body: copy_email_address,
+            id: "copy_email_address_modal",
+            html_submit_button: "Copy address",
+            on_click: () => {},
+            close_on_submit: true,
+        });
+        if (address.includes("show-sender")) {
+            $("#show-sender").prop("checked", true);
+        }
+        if (address.includes("include-footer")) {
+            $("#include-footer").prop("checked", true);
+        }
+        if (address.includes("include-quotes")) {
+            $("#include-quotes").prop("checked", true);
+        }
+        if (address.includes("prefer-html")) {
+            $("#prefer-html").prop("checked", true);
+        }
```

Comment:
Do we need to set the checkbox values here. I think we always have `show-sender` in the email shown here by default and not other values. So, we can just check the `show-sender` checkbox if we can confirm that only `show-sender` is present in the email shown here by default.

Reply:
Yes, it's always just `show-sender`